x-pack/plugins/uptime/public/apps/plugin.ts

```diff
@@ -56,16 +71,32 @@ export class UptimePlugin implements Plugin<void, void, ClientPluginsSetup, Clie
         );
 
         const { element } = params;
+
         const libs: UMFrontendLibs = {
-          framework: getKibanaFrameworkAdapter(coreStart, plugins, corePlugins),
+          framework: getKibanaFrameworkAdapter(coreStart, plugins, corePlugins, self._store),
         };
         libs.framework.render(element);
         return () => {};
       },
     });
   }
 
-  public start(_start: CoreStart, _plugins: {}): void {}
+  public start(start: CoreStart, _plugins: {}): void {
+    kibanaService.core = start;
+
+    alertTypeInitializers.forEach(init => {
+      const alertInitializer = init({
+        autocomplete: this._data!.autocomplete,
```

Comment:
This may not work correctly in a future release. We plan to lock down the `setup` APIs to only be callable during the `setup` phase. If you just need `data.autocomplete.getQuerySuggestions`, that is also available on `start`.

Reply:
Yeah I guess in general, I'm not sure why these initializers are called in `start` rather than setup? Seems like it all depends on setup APIs?

Reply:
the can perhaps be moved inside component by using useKibana hook.

Reply:
@joshdover can you have a look at https://github.com/elastic/kibana/pull/66054/commits/e5a1eb31c868ecf4d06e05f3442d4a5091ceedc2 when you've got a chance? I've moved the registration back to `setup` and removed instance fields for setup APIs.

Reply:
Great, that looks better 👍

Reply:
@joshdover I'm actually actively working on resolving more issues with this branch, but they're related to alerting. One thing we need is `core.services.application.getUrlForApp` from `CoreStart`. I've moved this back to `start`, but `triggers_actions_ui` is available on the start plugins param, and `autocomplete` is available as well (like you've mentioned).

I'm not referencing anything from `setup` now though. Can you verify this is still ok? I think once we've ironed that out it should be the end of platform-specific concerns. https://github.com/elastic/kibana/pull/66054/commits/bf16f6ad9cad517cfe2d5cd6e3eb978faa3b8dc1

Reply:
I wonder if we should reconsider how this whole thing is designed. Why is that the Redux store needs to exist at all? Is it possible we could decouple the UI from the store to make it more portable?